Classes/MapEditor/MapEditor.cs

```diff
@@ -382,33 +413,34 @@ public class MapEditor {
             // beat has been changed
             if (currentBeat != null){
 
-                // use current values for defaultBeats calculation
-                defaultBeats = GetDefaultBeats(currentBeat.gridDivision , currentBeat.BPM);
+                // calculate the new length factor
+                double scaleFactor = globalBPM / currentBeat.BPM;
+
+                // calculate the length of a single beat based on the new length
+                defaultBeats = scaleFactor / currentBeat.gridDivision;
 
-                // calculate time difference between old beat and start time
+                // calculate time difference between old beat and start float
                 double differenceDefaultNew = Math.Floor(currentBeat.globalBeat / defaultBeats) * defaultBeats;
 
                 // calculate offset
                 offset = currentBeat.globalBeat - differenceDefaultNew;
 
             }
-            double newBeat = Math.Floor(n.beat / defaultBeats) * defaultBeats;
+            double newBeat = Math.Round(n.beat / defaultBeats) * defaultBeats;
+
             newBeat += offset;
 
-            if (newBeat != n.beat) {
-                Note newNote = new Note(newBeat, n.col);
-                notesToAdd.Add(newNote);
-                notesToRemove.Add(n);
+            // no changes done? or its a step jump? do nothing.
+            if (Helper.DoubleApproxEqual(n.beat, newBeat) || Helper.DoubleApproxEqual(n.beat, newBeat - defaultBeats) ) {
```

Comment:
There's still issue with notes unselecting after quantization if they didn't move. 

The issue is now with this if and the interaction with UpdateNotes - since we don't touch those notes (they're neither in notesToAdd nor notesToRemove), UpdateNotes unselects them, but then doesn't reselect.

I think the easiest solution would be to just let quantization touch all notes, no matter if they move or not, the same as NoteTransforms interface.